File: 0643-maximum-average-subarray-i.c
#define MAX(x,y)(x > y ? x : y)
double findMaxAverage(int *nums, int numsSize, int k)
{
    double result = -INFINITY;
    double buf = 0.0;
    
    int windowStart = 0;
    for (int windowEnd = 0; windowEnd < numsSize; windowEnd++) {
        buf += nums[windowEnd];
        if (windowEnd >= k - 1) {
            result = MAX(result, buf / (double)k);
            buf -= nums[windowStart];
            windowStart++;
        }
    }
    
    return result;
}

File: 0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        
        max_avg = -math.inf
        buf = 0
        
        windowStart = 0
        
        for windowEnd in range(len(nums)):
            buf += nums[windowEnd]
            if windowEnd >= k - 1:
                max_avg = max(max_avg, buf / k)
                buf -= nums[windowStart]
                windowStart += 1
            
        return max_avg
